pkg/features/kube_features.go

```diff
@@ -938,6 +938,11 @@ const (
 	// Enables the image volume source.
 	ImageVolume featuregate.Feature = "ImageVolume"
 
+	// owner: @sreeram-venkitesh
+	//
+	// Denies pod admission if static pods reference other API objects.
+	StaticPodStrictValidation featuregate.Feature = "StaticPodStrictValidation"
```

Comment:
"strict" is pretty ambiguous ... let's target this specifically at static pods referencing API objects, so call this "PreventStaticPodAPIReferences" or something similar.

That will also let us make use of the same gate in the kubelet *and* in NodeRestriction admission